### PR TITLE
Add use cases. Refactor WebIDL.

### PR DESCRIPTION
**Pull request by zolkis**
WebIDL was updated based on comments in #3.
Added some of the use cases mentioned in #6.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -267,20 +267,23 @@
...
           <li>Add a property definition to the <a>Thing</a>.</li>
           <li>Add an event definition to the <a>Thing</a>.</li>
           <li>Add an action definition to the <a>Thing</a>.</li>
-          <li>Attach semantic information to an Action</li>
-          <li>Attach semantic information to a Property</li>
-          <li>Attach semantic information to an Event</li>
-          <li>Start the exposed <a>Thing</a> in order to process external requests.</li>
-          <li>Modify the handling of an external request to retrieve the <a>Thing Description</a>.</li>
-          <li>React on a property change</li>
+          <li>Register an action triggered by external retrieve requests on the <a>Thing Description</a>.</li>
```
Comment (h0ru5) [inline index.html]: to avoid confusion with the interacion pattern "action" (of a thing) , could we say "a callback" or "logic"?

Comment (h0ru5) [inline index.html]: to avoid confusion with the interacion pattern "action" (of a thing) , could we say "a callback" or "logic"?

**Comment by zolkis** [comment]
What about the following:
- for request handlers we use single callback (to handle requests like retrieve TD, run Action, add Event listener, etc)
- for hooks we use built-in events, like `on("actionrequest")`, `on("propertychangerequest") to which multiple listeners can be attached. In the use case, we can use the term "observe", like "Observe requests for invoking an action", and "Observe a request for property change". The effect is that after a property is changed or an action is invoked, listeners are notified in order to update their business logic.

However, I still fail the understand the use case for the hooks. These hooks are only used by the same script that defines the server Thing. Why cannot the logic that goes into the hooks go directly into the request handlers?

**Comment by zolkis** [comment]
I have reformulated the use cases and moved all observation use cases to consumed Thing. The misunderstanding probably came from the fact that ExposedThing also implements ConsumedThing and in the Current Practices they are just dumped together.

**Comment by zolkis** [comment]
@knimura @h0ru5 could you review again, please

**Comment by zolkis** [comment]
Added request handlers to `ExposedThing` and consolidated use of dictionaries for properties, actions and events.

**Comment by zolkis** [comment]
Based on [this comment](https://github.com/w3c/wot-scripting-api/issues/6#issuecomment-286037138) I am merging for now. If there is feedback, let's raise them later.

**Pull request merged**